utilities/update_links.py

```diff
@@ -30,49 +40,78 @@ def _get_files(self):
+                print(f'Filename: {x}, occurences: {old_string_count}')
+                data.append([x, old_string_count])
+            if old_string_count > 0:
+                file_count += 1
+        print(f"Number of files to change: {file_count}")
+        summary = pd.DataFrame(data, columns = cols)
+        return summary
 
 
 
     def _change_string(self, filename):
-            with open(filename, 'r', encoding = 'utf8') as search_file:
-                x = search_file.read()
-                if old_string not in x:
-                    print(f'{self.old_string} not found in {filename}')
-                    search_file.close()
-            with open(filename, 'w', encoding = 'utf8') as file_to_change:
-                print(f'Changing {self.old_string} to {self.new_string} in {filename}')
-                x = x.replace(self.old_string, self.new_string)
-                file_to_change.write(x)
-                file_to_change.close()
+        '''
+        Changes string from old_string to new_string
+        '''
+        with open(filename, 'r', encoding = 'utf8') as search_file:
+            x = search_file.read()
+            if old_string not in x:
+                print(f'{self.old_string} not found in {filename}')
+                search_file.close()
+        with open(filename, 'w', encoding = 'utf8') as file_to_change:
+            print(f'Changing {self.old_string} to {self.new_string} in {filename}')
+            x = x.replace(self.old_string, self.new_string)
+            file_to_change.write(x)
+            file_to_change.close()
 
 
     def find_and_replace(self):
-
+        '''
+        Loops through each file from get_files() and uses change_string() on it.
+        '''
         file_list = self._get_files()
         for x in file_list:
-            print(f'opening {x}')
+            #print(f'opening {x}')
             self._change_string(x)
 
-   
 
-docs = FindReplace(filepath, extensions, old_string, new_string)
-docs.count_replacements()
-docs.find_and_replace()
-docs.count_replacements()
+
+def search_and_replace(filepath, extensions, old_string, new_string):  
+    '''
+    Wrapper function.
+    Counts number of files, and occurences within these files of the old_string.
+    Replaces old_string with new_string in all files.
+    Recounts the number of files, and occurences of old_string. This should now be 0.
+
+    Args:
+        filepath (string): filepath to directory you want to search through.
+        extensions (tuple containing multiple strings): extensions of the files you want to change.
+        old_string (string): string to search for which requires changing.
+        new_string (string): string you want to change to.
+    '''
+
+    docs = FindReplace(filepath, extensions, old_string, new_string)
+    print(docs.count_replacements())
+    docs.find_and_replace()
+    print(docs.count_replacements())
+
+
+
+
+#wrapper function to run
+search_and_replace(filepath, extensions, old_string, new_string)
```

Comment:
add this into the documentation for the wrapper function as example usage

Reply:
added under Example